store/postgres/tests/relational_bytes.rs

```diff
@@ -0,0 +1,311 @@
+    Value, BLOCK_NUMBER_MAX,
+};
+use graph_store_postgres::layout_for_tests::Layout;
+
+use test_store::*;
+
+const THINGS_GQL: &str = "
+    type Thing @entity {
+        id: ID!
+        name: String!
+    }
+";
+
+const SCHEMA_NAME: &str = "layout";
+
+lazy_static! {
+    static ref THINGS_SUBGRAPH_ID: SubgraphDeploymentId =
+        SubgraphDeploymentId::new("things").unwrap();
+    static ref LARGE_INT: BigInt = BigInt::from(std::i64::MAX).pow(17);
+    static ref LARGE_DECIMAL: BigDecimal =
+        BigDecimal::one() / LARGE_INT.clone().to_big_decimal(BigInt::from(1));
+    static ref BYTES_VALUE: H256 = H256::from(hex!(
+        "e8b3b02b936c4a4a331ac691ac9a86e197fb7731f14e3108602c87d4dac55160"
+    ));
+    static ref BYTES_VALUE2: H256 = H256::from(hex!(
+        "b98fb783b49de5652097a989414c767824dff7e7fd765a63b493772511db81c1"
+    ));
+    static ref BYTES_VALUE3: H256 = H256::from(hex!(
+        "977c084229c72a0fa377cae304eda9099b6a2cb5d83b25cdf0f0969b69874255"
+    ));
+    static ref BEEF_ENTITY: Entity = {
+        let mut entity = Entity::new();
+        entity.set("id", "deadbeef");
+        entity.set("name", "Beef");
+        entity.set("__typename", "Thing");
+        entity
+    };
+}
+
+/// Removes test data from the database behind the store.
+fn remove_test_data(conn: &PgConnection) {
+    let query = format!("drop schema if exists {} cascade", SCHEMA_NAME);
+    conn.batch_execute(&query)
+        .expect("Failed to drop test schema");
+}
+
+fn insert_entity(conn: &PgConnection, layout: &Layout, entity_type: &str, entity: Entity) {
+    let key = EntityKey {
+        subgraph_id: THINGS_SUBGRAPH_ID.clone(),
+        entity_type: entity_type.to_owned(),
+        entity_id: entity.id().unwrap(),
+    };
+    let errmsg = format!("Failed to insert entity {}[{}]", entity_type, key.entity_id);
+    layout.insert(&conn, &key, &entity, 0).expect(&errmsg);
+}
+
+fn insert_thing(conn: &PgConnection, layout: &Layout, id: &str, name: &str) {
+    let mut thing = Entity::new();
+
+    thing.insert("id".to_owned(), Value::String(id.to_owned()));
+    thing.insert("name".to_owned(), Value::String(name.to_owned()));
+
+    insert_entity(conn, layout, "Thing", thing);
+}
+
+fn create_schema(conn: &PgConnection) -> Layout {
+    let schema = Schema::parse(THINGS_GQL, THINGS_SUBGRAPH_ID.clone(), IdType::Bytes).unwrap();
+
+    let query = format!("create schema {}", SCHEMA_NAME);
+    conn.batch_execute(&*query).unwrap();
+
+    let layout = Layout::create_relational_schema(
+        &conn,
+        SCHEMA_NAME,
+        schema.id_type,
+        THINGS_SUBGRAPH_ID.clone(),
+        &schema.document,
+    )
+    .expect("Failed to create relational schema");
+
+    layout
+}
+
+fn scrub(entity: &Entity) -> Entity {
+    let mut scrubbed = Entity::new();
+    // merge has the sideffect of removing any attribute
+    // that is Value::Null
+    scrubbed.merge(entity.clone());
+    scrubbed
+}
+
+macro_rules! assert_entity_eq {
```

Comment:
I have nothing against macros, but this to me looks like it could also just be a regular function; unless I'm missing something?